### DISCUSSION TITLE
How to see "Processed Raw" in the source next to "Raw"

### DISCUSSION THREAD

**Discussion by Graziano-Calella** [nextgenhealthcare/connect#6336] [Q&A] [answered]
In the channel source I get a message like this:

""fake message example""
`<env:Envelope xmlns:env="http://pippo.org/2007/02/soap-envelope">
    <env:Header/>
    <env:Body>
        <MSG_HL7>PEhMQSUQuMy44LjI3LjY+PC9PQlI+PE9CUi4zNi8+PC9PQlI+PC9ITDdNZXNzYWdlPg==</MSG_HL7>
    </env:Body>
</env:Envelope>`

In the preprocessor I have this code:

`var message_xml = new XML(message);
message = new java.lang.String(FileUtil.decode(message_xml.*::['Body']['MSG_HL7'].toString())).toString();

return message;`

The script as final output returns the decoded base64 message which contains a hl7 XML message.

At this point, if I set the inbound of the XML type source there are no errors, but I don't see the "Processed Raw" tab appear.
While if I set the inbound of the HL7 pipe v2.x type source, the source fails (and doesn't go forward in the destinations), but I finally see the "Processed Raw" tab with the transformation from the enveloped message that I received, to the one decoded in XML.

Furthermore, when the source fails, I also see a third tab, that of the "transformer" in addition to "raw" and "Processed Raw", with a completely wrong value, that is, the opening and closing tags of the xml are messed up.

What I would like is to see the "Processed Raw" in the source and that it doesn't fail obviously, so that I can work with the destinations later.
How can I do it?

I attach some images that may be helpful

![Screenshot (33)](https://github.com/user-attachments/assets/0e79dbe2-0781-4bf5-9a76-6b084688942e)

![Screenshot (34)](https://github.com/user-attachments/assets/f28a509f-91db-4cd5-9252-29c808ff76c2)

![Screenshot (32)](https://github.com/user-attachments/assets/6108564e-7352-49f0-ae8f-d83a6cd2ac95)

![Screenshot (35)](https://github.com/user-attachments/assets/19003191-b050-4913-bb28-1c941a0bda99)

**Comment by pacmano1** [comment]
Moving your post to the Q and A section.  

Please also use code formatting:

```JS
var message_xml = new XML(message);
message = new java.lang.String(FileUtil.decode(message_xml.*::['Body']['MSG_HL7'].toString())).toString();

return message;
```

**Comment by pacmano1** [comment]
@Graziano-Calella I am not exactly sure what you are asking here.

Is this actually a web service listener that receives an HL7 message (in XML itself) that is base64 encoded?

**Reply by Graziano-Calella** [reply]
yes exactly, but regardless of what I get, I would like to better understand how to activate, visually in the dashboard, the "Processed Raw".

**Reply by pacmano1** [reply, marked as answer]
It only activates that radio button if the preprocessor runs (i.e. has something other than return message in the script)

**Reply by Graziano-Calella** [reply]
Mhhh, can you by any chance create me a little script that will show me how to do it or can you send me the export of the channel, please?

**Reply by pacmano1** [reply]
<img width="305" alt="image" src="https://github.com/user-attachments/assets/105f52ed-5c36-4ec4-9250-98cccea8d6e6">
